### PR TITLE
Read mirror auth credentials from the target namespace

### PR DESCRIPTION
**Pull request by ricardomaraschini**
**We don't know if we want this yet but for sake of not losing it I am opening a PR**

Instead of having a central place where we configure the mirror registry we allow now users to configure only the mirror registry URL and when importing images we attempt to read secrets for the configured registry from within the target namespace (target namespace is where the are going to store the Image object).

### CODE REVIEW

File: services/sysctx.go [hunk 1]
```diff
@@ -16,74 +16,27 @@ package services
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
 	"os"
 
-	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (otaviof) [inline services/sysctx.go:22]: nit: usually that would be named "corev1", instead of "v1" only.

File: cmd/kubectl-image/import.go [hunk 2]
```diff
@@ -54,16 +55,32 @@ var imageimport = &cobra.Command{
 			return err
 		}
 
-		mirror, err := c.Flags().GetBool("mirror")
+		mirror, err := c.Flags().GetString("mirror")
```
Comment (imjasonh) [inline cmd/kubectl-image/import.go:58]: Why change this from a bool to a string that can only be `"true"` or `"false"`?

File: infra/imagestore/registry.go [hunk 3]
```diff
@@ -50,93 +51,127 @@ type Registry struct {
...
-		ctx, i.polctx, dstref, ref, &imgcopy.Options{
-			SourceCtx: i.regctx,
-		},
-	)
-	if err != nil {
-		cleanup()
-		return nil, nil, fmt.Errorf("unable to copy image: %w", err)
+	var err error
+	var cleanup CleanFn
+	var destref types.ImageReference
```
Comment (imjasonh) [inline infra/imagestore/registry.go:147]: These variables don't escape the `for` loop scope, so they could just be instantiated with `destref, cleanup, err := i.NewLocalReference` on line 150, right?

File: cmd/kubectl-image/import.go (hunk 2 above)
Comment (ricardomaraschini) [reply cmd/kubectl-image/import.go:58]: Let me see if I can explain this: `mirror` and `insecure-source` can be set to one of these values: "true", "false" or "undefined". If "undefined" then we use whatever is defined in the Image as the default for the ImageImport. If it is "true" or "false" instead then we "overwrite" (for that specific import) the values from the "parent Image". Does that make sense?

File: infra/imagestore/registry.go (hunk 3 above)
Comment (ricardomaraschini) [reply infra/imagestore/registry.go:147]: nice catch, yes.

File: cmd/kubectl-image/import.go (hunk 2 above)
Comment (ricardomaraschini) [reply cmd/kubectl-image/import.go:58]: For sake of completeness: https://github.com/shipwright-io/community/pull/57#discussion_r801516122

**Comment by ricardomaraschini** [comment]
Replaced by https://github.com/shipwright-io/image/pull/21

**Pull request closed without merging**